### PR TITLE
11 support for custom flags

### PR DESCRIPTION
**Pull request by daniaki**
PR for issue #11 

@danielreti Could you please review the React components and Redux additions? (in the `/assets/` folder)

Screenshots
------------
Screenshot 2023-01-11 at 10 02 35 am
Screenshot 2023-01-11 at 10 02 46 am
Screenshot 2023-01-11 at 10 02 55 am
Screenshot 2023-01-11 at 10 03 26 am

### CODE REVIEW

**Review by danielreti — APPROVED** [review]
React bits look good to me - nice work with the classes

**Review by illusional — APPROVED** [review]
Python / Django side looks pretty good, nothing obvious I see

File: curation_portal/management/commands/create_custom_flag.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
+from django.core.management import BaseCommand
+
+from curation_portal.models import CustomFlag
+
+
+class Command(BaseCommand):
```
[the remark this answers is not in the document]
Comment (daniaki) [reply curation_portal/management/commands/create_custom_flag.py:6]: It is an endpoint at `/api/custom_flag/create`, but I also made it a command for local dev testing

File: curation_portal/serializers.py [hunk 2]
```diff
@@ -151,6 +156,92 @@ def create(self, validated_data):
...
+        return value  # return original value for further validation
+
+    def validate_shortcut(self, value):
+        shortcut = value
+        if shortcut:
+            shortcut = str(value).upper()
+
+        if shortcut.upper()[0] in ("R", "S"):
+            raise ValidationError(
+                "Shortcut cannot start with 'S' since it is used as the shortcut to save a "
```
[the remark this answers is not in the document]
Comment (daniaki) [reply curation_portal/serializers.py:183]: Oops, "R" isn't supposed to be in the tuple anymore. Forgot to remove it.

**Pull request merged**